src/lib/use-toggle.ts

```diff
@@ -0,0 +1,18 @@
+import { useCallback, useState } from 'react'
+
+/**
+ * Simple util to quickly handle on/off states.
+ */
+const useToggle = (initialState: boolean = false): [boolean, () => void] => {
+  const [isTrue, setIsTrue] = useState(initialState)
+
+  // use callback otherwise it would yield unexpected results in promises
```

Comment:
Interesting, curious how you learned about this. Is it specific to `useToggle` or is this something you always need to worry about with `useState`?